### DISCUSSION TITLE
Xml handled differently between test and prod servers

### DISCUSSION THREAD

**Discussion by Mistaken4now** [nextgenhealthcare/connect#5311] [Q&A] [answered]
this line of code work in production on our prod server
this line of code
msg['PID']['PID.3']['PID.3.1'] = msg['PID']['PID.2'][2]['PID.2.1'].toString();

ie
**D|1|E1507522^^^EE^EPI~ 2186469^^^CORP^EE~ 3412407^^^UMRN^MR|350626^^^PIT^MR|**

3.1 gets set to 3412407

on our test server the same line errors out the process

397: // Keep MRN (PID-3.1) and remove all the subsequent fields (PID-3.2 ~ PID-3.x).
398: msg['PID']['PID.3']['PID.3.1'] = msg['PID']['PID.2'][2]['PID.2.1'].toString();
399: /*
400: var mrn = msg['PID']['PID.3']['PID.3.1'].toString();
401: msg['PID']['PID.3'] = "";
402: if(msg['MSH']['MSH.4']['MSH.4.1'].toString() == "BFT"){
LINE NUMBER:	398
DETAILS:	TypeError: Cannot read property "PID.2.1" from undefined

is there a setting we need to turn on in test, to have it process 'arrays'  with in fields?

**Comment by siddharthsharma9258** [comment, marked as answer]
Parse field repetitions is an option under data types properties for HL7, which might be disabled(!) in test, that is why there could be a difference between test and prod.

Also, Your current piece of code _always_ assume that PID.2  would be in the format that you gave. But the correct way is to first check the size..

`if(msg['PID']['PID.2'].length() >= 1) {
//your code here}
`

**Reply by Mistaken4now** [reply]
thank you sir, those check boxes got us fixed up.
